test/cron_test.rb

```diff
@@ -0,0 +1,132 @@
+require_relative 'test_helper'
+require 'rack'
+require 'json'
+require 'open-uri'
+
+SingleCov.not_covered!
+
+describe 'CRON' do
+  class FakeServer
+    TEST_ENDPOINT = "/__ping__".freeze
+
+    def self.open(port, replies)
+      server = new(port, replies)
+      server.boot
+      yield server
+    ensure
+      server.shutdown
+    end
+
+    def initialize(port, replies)
+      @port = port
+      @replies = replies.merge(TEST_ENDPOINT => 'PONG')
+    end
+
+    def boot
+      Thread.new do
+        Rack::Handler::WEBrick.run(
+          self,
+          Port: @port,
+          Logger: WEBrick::Log.new("/dev/null"),
+          AccessLog: []
+        ) { |s| @server = s }
+      end
+    end
+
+    def wait
+      loop do
+        begin
+          break if URI.parse("http://localhost:#{@port}#{TEST_ENDPOINT}").read
+        rescue
+          nil
+        end
+      end
+    end
+
+    def call(env)
+      path = env.fetch("PATH_INFO")
+      unless reply = @replies[path]
+        puts "ERROR: Missing reply for path #{path}" # kubeclient does not show current url when failing
+        raise
+      end
+      [200, {}, [reply.to_json]]
+    end
+
+    def shutdown
+      @server.shutdown if @server
+    end
+  end
+
+  let(:br) { Bundler.root.to_s }
+  let(:vault_replies) do
+    {
+      '/auth/token/create' =>
+        {
+          "auth": {
+            "client_token": "ABCD",
+            "policies": ["web", "stage"],
+            "metadata": {"user": "armon"},
+            "lease_duration": 3600,
+            "renewable": true,
+          }
+        },
+      '/auth' => {}
+
+    }
+  end
+  let(:kuber_replies) do
+    {
+      '/api/v1/namespaces/default/secrets/vaultauth' => {},
+    }
+  end
+
+  describe "failure cases" do
+    before do
+      ENV["VAULT_PORT"] = "8201"
+    end
+    after do
+      ENV.delete("VAULT_PORT")
+    end
+
+    it "fails w/o the server listening" do
+      e = assert_raises RuntimeError do
+        sh "#{br}/bin/create_k8s_token.sh -v http://localhost:9090 -t faketoken -k #{br}/test/fixutes/kube_config"
+      end
+      e.message.must_include("FAILED could not get token for secret puller")
+    end
+
+    it "fails with missing token" do
+      e = assert_raises RuntimeError do
+        sh "#{br}/bin/create_k8s_token.sh -v https://localhost:1234h"
+      end
+      e.message.must_include("VAULT_TOKEN missing")
+    end
+
+    it "fails with missing vault address" do
+      e = assert_raises RuntimeError do
+        sh "#{br}/bin/create_k8s_token.sh -t faketoken"
+      end
+      e.message.must_include("VAULT_ADDR missing")
+    end
+  end
+
+  around do |test|
```

Comment:
needs to be in the top ... or in it's own describe ... this is used for everything atm